reconciletags/reconciletags_test.py

```diff
@@ -60,45 +64,80 @@ def setUp(self):
                        'repository': _REPO,
                        'images': [{'digest': _DIGEST1, 'tag': _TAG1}]}]}
 
-    def test_reconcile_tags(self):
-        with mock.patch('subprocess.check_output',
-                        return_value=_LIST_RESP) as mock_output:
+    @patch('containerregistry.client.v2_2.docker_session.Push')
+    @patch('containerregistry.client.v2_2.docker_image.FromRegistry')
+    def test_reconcile_tags(self, mock_from_registry, mock_push):
+        fake_base = mock.MagicMock()
+        fake_base.tags.return_value = [_TAG1]
+
+        mock_img = mock.MagicMock()
+        mock_img.__enter__.return_value = fake_base
+        mock_from_registry.return_value = mock_img
+        mock_push.return_value = docker_session.Push()
+
+        with mock.patch('reconciletags.logging.debug') as mock_output:
+
             self.r.reconcile_tags(self.data, False)
-            mock_output.assert_any_call([_GCLOUD_CONFIG], shell=True)
-            mock_output.assert_any_call([_GCLOUD_LIST], shell=True)
-            mock_output.assert_any_call([self._gcloudAdd(_DIGEST1, _TAG1)],
-                                        shell=True)
-
-    def test_dry_run(self):
-        with mock.patch('subprocess.check_output',
-                        return_value=_LIST_RESP) as mock_output:
+            logging_debug_output = [call[1][0] for
+                                    call in mock_output.mock_calls]
+
+            assert mock_from_registry.called
+            assert mock_push.called
+
+            self.assertIn(self._tagging(_DIGEST1, _TAG1), logging_debug_output)
+            self.assertIn(_EXISTING_TAGS, logging_debug_output)
+
+    @patch('containerregistry.client.v2_2.docker_session.Push')
+    @patch('containerregistry.client.v2_2.docker_image.FromRegistry')
+    def test_dry_run(self, mock_from_registry, mock_push):
+        mock_from_registry.return_value = docker_image.FromRegistry()
+        mock_push.return_value = docker_session.Push()
+
+        with mock.patch('reconciletags.logging.debug') as mock_output:
```

Comment:
You probably don't need to patch or have assertions on the logging calls, that's a bit overkill.

Sorry for not noticing this earlier!

Reply:
no worries! so it's enough to make sure the mocked functions are called?